src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexCompiler.cs

```diff
@@ -64,6 +64,12 @@ internal abstract class RegexCompiler
         private static readonly MethodInfo s_debugWriteLine = typeof(Debug).GetMethod("WriteLine", new Type[] { typeof(string) })!;
 #endif
 
+        /// <summary>
+        /// The max recursion depth used for computations that can recover for not walking the entire node tree.
+        /// This is used to avoid stack overflows on degenerate expressions.
+        /// </summary>
+        private const int MaxRecursionDepth = 20;
```

Comment:
Do we have sufficient test coverage of such degenerate expressions?

Reply:
I'll add one if we don't.